packages/datepickers/src/Datepicker.tsx

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useEffect, useReducer, useCallback, FunctionComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Manager, Popper, Reference } from 'react-popper';
+import isSameDay from 'date-fns/isSameDay';
+import isValid from 'date-fns/isValid';
+import isBefore from 'date-fns/isBefore';
+import { isRtl, withTheme } from '@zendeskgarden/react-theming';
+import { KEY_CODES } from '@zendeskgarden/container-utilities';
+
+import { StyledMenu } from './styled';
+import {
+  getRtlPopperPlacement,
+  getPopperPlacement,
+  GARDEN_PLACEMENT
+} from './utils/garden-placements';
+import Calendar from './components/Calendar';
+import { datepickerReducer, retrieveInitialState } from './utils/reducer';
+import { DatepickerContext } from './utils/useDatepickerContext';
+
+/**
+ * Parse string input value using current locale and date formats
+ */
+function parseInputValue({ inputValue }: { inputValue: string }): Date {
+  const MINIMUM_DATE = new Date(1001, 0, 0);
+  const tryParseDate = new Date(inputValue);
+
+  if (
+    !isValid(tryParseDate) ||
+    isSameDay(tryParseDate, MINIMUM_DATE) ||
+    isBefore(tryParseDate, MINIMUM_DATE)
+  ) {
+    return new Date(NaN);
+  }
+
+  return tryParseDate;
+}
+
+export interface IDatepickerProps {
+  /**
+   * The selected date to display
+   */
+  value?: Date;
+  /**
+   * Returns the parsed date
+   */
+  onChange?: (date: Date) => void;
+  /**
+   * Allows customization of date formatting within input.
+   */
+  formatDate?: (date: Date) => string;
+  /**
+   * Accepts [all valid Intl locales](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation)
+   */
+  locale?: string;
+  /**
+   * Minimum value. Users are still able to manually enter a date below this value.
+   */
+  minValue?: Date;
+  /**
+   * Maximum value. Users are still able to manually enter a date above this value.
+   */
+  maxValue?: Date;
+  /**
+   * Show compact styling
+   */
+  small?: boolean;
+  /**
+   * Override default date parsing
+   */
+  customParseDate?: (inputValue: string) => Date;
+  /**
+   * The ref key used to position the dropdown
+   */
+  refKey?: string;
+  /** Locale-aware placement for the dropdown element */
+  placement?: GARDEN_PLACEMENT;
+  /** Display an arrow to the reference element */
+  arrow?: boolean;
+  /**
+   * Passes options to [Popper.JS Instance](https://github.com/FezVrasta/popper.js/blob/master/docs/_includes/popper-documentation.md#new-popperreference-popper-options)
+   */
+  popperModifiers?: any;
+  /**
+   * Show open animations
+   */
+  animate?: boolean;
```

Comment:
Should this show a `defaultProps` of `true`?

Reply:
It looks like the props-parser for Typescript doesn't take `defaultProps` into account https://github.com/styleguidist/react-docgen-typescript/issues/154

I've added the `@defualt` JSDoc comments to make this visible for now